presto-main/src/main/java/com/facebook/presto/execution/WarningSink.java

```diff
@@ -15,11 +15,11 @@
 
 import com.facebook.presto.spi.WarningCode;
 
-import java.util.Collection;
+import java.util.List;
 
 public interface WarningSink
```

Comment:
This interface is interesting. It's a sink according to its name, but at the same time we can read from it with `getWarnings`. I think we should rework this API around `Consumer<PrestoWarning>` and `Supplier<List<PrestoWarning>>`. 

How about we rename this as `WarningCollector` (to get rid of the readable sink confusion) and implement both `Consumer<PrestoWarning>` and `Supplier<List<PrestoWarning>>`?

```
public interface WarningCollector
        extends Consumer<PrestoWarning>, Supplier<List<PrestoWarning>>
{
}
```

Reply:
+1, I think WarningsCollector is a better name. I wouldn't worry about making it implement Consumer/Supplier, though, but it depends on how it's being used elsewhere.